### PR TITLE
docs(automl): fix assorted typos/bugs doc strings

### PR DESCRIPTION
**Pull request by lwander**

### CODE REVIEW

File: automl/google/cloud/automl_v1beta1/tables/tables_client.py [hunk 1]
```diff
@@ -634,7 +634,8 @@ def delete_dataset(
 
         Returns:
             A :class:`~google.cloud.automl_v1beta1.types._OperationFuture`
-            instance.
+            instance. See the documentation here:
+            https://googleapis.dev/python/google-api-core/latest/operation.html
```
Comment (busunkim96) [inline automl/google/cloud/automl_v1beta1/tables/tables_client.py]: If the return type is listed in the public docs, you can list it here (e.g., `google.api_core.retry.Retry`,  `google.api_core.operation.Operation`) and the doc generator will make a link. See [examples](https://googleapis.dev/python/automl/latest/gapic/v1beta1/tables.html).

File: automl/google/cloud/automl_v1beta1/tables/tables_client.py [hunk 2]
```diff
@@ -1172,6 +1175,13 @@ def update_column_spec(
                 to provide either a way to lookup the source dataset (using one
                 of the `dataset*` kwargs), or the `table_spec_name` of the
                 table this column belongs to.
+            type_code (Optional[str]):
+                The desired 'type_code' of the column. For more information
+                on the available types, please see the proto definition:
+                https://github.com/googleapis/google-cloud-python/blob/bfb4da8542981d2eedffe20f64e87ab528a17592/automl/google/cloud/automl_v1beta1/proto/data_types.proto
```
[the remark this answers is not in the document]
Comment (lwander) [reply automl/google/cloud/automl_v1beta1/tables/tables_client.py]: Good call, done

File: automl/google/cloud/automl_v1beta1/tables/tables_client.py [hunk 3]
```diff
@@ -1172,6 +1175,13 @@ def update_column_spec(
                 to provide either a way to lookup the source dataset (using one
```
[the remark this answers is not in the document]
Comment (lwander) [reply automl/google/cloud/automl_v1beta1/tables/tables_client.py]: Reordered for this funciton -- the others will have to wait we have people depending on the non-cosmetic doc fixes in this PR.

File: automl/google/cloud/automl_v1beta1/tables/tables_client.py (hunk 1 above)
Comment (lwander) [reply automl/google/cloud/automl_v1beta1/tables/tables_client.py]: Updated to use above return format & `google.api_core.operation.Operation` class

**Comment by tseaver** [comment]
@lwander `nox -s lint` is failing:

```
google/cloud/automl_v1beta1/tables/tables_client.py:635:17: W291 trailing whitespace
google/cloud/automl_v1beta1/tables/tables_client.py:637:66: W291 trailing whitespace
google/cloud/automl_v1beta1/tables/tables_client.py:741:17: W291 trailing whitespace
google/cloud/automl_v1beta1/tables/tables_client.py:743:66: W291 trailing whitespace
google/cloud/automl_v1beta1/tables/tables_client.py:848:17: W291 trailing whitespace
google/cloud/automl_v1beta1/tables/tables_client.py:850:66: W291 trailing whitespace
google/cloud/automl_v1beta1/tables/tables_client.py:2171:17: W291 trailing whitespace
google/cloud/automl_v1beta1/tables/tables_client.py:2173:66: W291 trailing whitespace
google/cloud/automl_v1beta1/tables/tables_client.py:2301:17: W291 trailing whitespace
google/cloud/automl_v1beta1/tables/tables_client.py:2303:66: W291 trailing whitespace
google/cloud/automl_v1beta1/tables/tables_client.py:2485:17: W291 trailing whitespace
google/cloud/automl_v1beta1/tables/tables_client.py:2487:66: W291 trailing whitespace
google/cloud/automl_v1beta1/tables/tables_client.py:2557:17: W291 trailing whitespace
google/cloud/automl_v1beta1/tables/tables_client.py:2559:66: W291 trailing whitespace
google/cloud/automl_v1beta1/tables/tables_client.py:2759:17: W291 trailing whitespace
google/cloud/automl_v1beta1/tables/tables_client.py:2761:66: W291 trailing whitespace
```

**Pull request merged**